src/Protocol/DFRN.php

```diff
@@ -1368,6 +1369,43 @@ public static function deliver($owner, $contact, $atom, $dissolve = false)
 		return intval($res->status);
 	}
 
+	/**
+	 * @brief Delivers items to the contacts via the Diaspora transport layer
+	 *
+	 * @param array $owner    Owner record
+	 * @param array $contact  Contact record of the receiver
+	 * @param array $items    Items that will be transmitted
+	 *
+	 * @return int HTTP Deliver status
+	 */
+	public static function buildAndTransmit($owner, $contact, $items)
+	{
+		$a = get_app();
+
+		// Currently disabled, at first we will not use the batch delivery
+		// $public_batch = !$items[0]['private'];
+		$public_batch = false;
+
+		$msg = DFRN::entries($items, $owner);
+
+		$fcontact = Diaspora::personByHandle($contact['addr']);
+		if (empty($fcontact)) {
+			logger("unable to find contact details");
+			return;
+		}
+
+		$envelope = Diaspora::buildMessage($msg, $owner, $contact, $owner['uprvkey'], $fcontact['pubkey'], $public_batch);
+
+		$dest_url = ($public_batch ? $fcontact["batch"] : $contact["notify"]);
+
+		$content_type = ($public_batch ? "application/magic-envelope+xml" : "application/json");
+
+		$ret = Network::post($dest_url, $envelope, ["Content-Type: ".$content_type]);
+
+		/// @ToDo: Add better treating of return codes
```

Comment:
`treatment` 😋

Reply:
"treating" exists as well :stuck_out_tongue: - (But currently not in this context?)

Reply:
Done